Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/jupyter_core/utils/__init__.py", line 173, in wrapped
    return loop.run_until_complete(inner)
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
generate_compass_dataset(g, output_directory, add_grade=True, energy_model="2016_TOYOTA_Camry_4cyl_2WD")      
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mImportError[0m                               Traceback (most recent call last)
Cell [0;32mIn[4], line 1[0m
[0;32m----> 1[0m [43mgenerate_compass_dataset[49m[43m([49m[43mg[49m[43m,[49m[43m [49m[43moutput_directory[49m[43m,[49m[43m [49m[43madd_grade[49m[38;5;241;43m=[39;49m[38;5;28;43;01mTrue[39;49;00m[43m,[49m[43m [49m[43menergy_model[49m[38;5;241;43m=[39;49m[38;5;124;43m"[39;49m[38;5;124;43m2016_TOYOTA_Camry_4cyl_2WD[39;49m[38;5;124;43m"[39;49m[43m)[49m      

File [0;32m/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/nrel/routee/compass/io/generate_dataset.py:85[0m, in [0;36mgenerate_compass_dataset[0;34m(g, output_directory, hwy_speeds, fallback, agg, add_grade, raster_resolution_arc_seconds, default_config, energy_model)[0m
[1;32m     83[0m [38;5;28;01mif[39;00m add_grade:
[1;32m     84[0m     log[38;5;241m.[39minfo([38;5;124m"[39m[38;5;124madding grade information[39m[38;5;124m"[39m)
[0;32m---> 85[0m     g1 [38;5;241m=[39m [43madd_grade_to_graph[49m[43m([49m
[1;32m     86[0m [43m        [49m[43mg1[49m[43m,[49m[43m [49m[43mresolution_arc_seconds[49m[38;5;241;43m=[39;49m[43mraster_resolution_arc_seconds[49m
[1;32m     87[0m [43m    [49m[43m)[49m
[1;32m     89[0m v, e [38;5;241m=[39m ox[38;5;241m.[39mgraph_to_gdfs(g1)
[1;32m     91[0m [38;5;66;03m# process vertices[39;00m

File [0;32m/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/nrel/routee/compass/io/utils.py:169[0m, in [0;36madd_grade_to_graph[0;34m(g, output_dir, resolution_arc_seconds)[0m
[1;32m    164[0m     downloaded_file [38;5;241m=[39m _download_tile(
[1;32m    165[0m         tile, output_dir[38;5;241m=[39moutput_dir, resolution[38;5;241m=[39mresolution
[1;32m    166[0m     )
[1;32m    167[0m     files[38;5;241m.[39mappend([38;5;28mstr[39m(downloaded_file))
[0;32m--> 169[0m g [38;5;241m=[39m [43mox[49m[38;5;241;43m.[39;49m[43madd_node_elevations_raster[49m[43m([49m[43mg[49m[43m,[49m[43m [49m[43mfiles[49m[43m)[49m
[1;32m    170[0m g [38;5;241m=[39m ox[38;5;241m.[39madd_edge_grades(g)
[1;32m    172[0m [38;5;28;01mreturn[39;00m g

File [0;32m/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/osmnx/elevation.py:129[0m, in [0;36madd_node_elevations_raster[0;34m(G, filepath, band, cpus)[0m
[1;32m    127[0m [38;5;28;01mif[39;00m rasterio [38;5;129;01mis[39;00m [38;5;28;01mNone[39;00m [38;5;129;01mor[39;00m gdal [38;5;129;01mis[39;00m [38;5;28;01mNone[39;00m:  [38;5;66;03m# pragma: no cover[39;00m
[1;32m    128[0m     msg [38;5;241m=[39m [38;5;124m"[39m[38;5;124mgdal and rasterio must be installed to query raster files[39m[38;5;124m"[39m
[0;32m--> 129[0m     [38;5;28;01mraise[39;00m [38;5;167;01mImportError[39;00m(msg)
[1;32m    131[0m [38;5;28;01mif[39;00m cpus [38;5;129;01mis[39;00m [38;5;28;01mNone[39;00m:
[1;32m    132[0m     cpus [38;5;241m=[39m mp[38;5;241m.[39mcpu_count()

[0;31mImportError[0m: gdal and rasterio must be installed to query raster files

